### PR TITLE
CDM-44: EventElement & EventGrid components

### PR DESCRIPTION
**Pull request by mariasbl**
- **Web**
image

- **Mobile**
   - Medium Screen
image
   - Small Screen
image

### Reviewers
- [ ] @danielaprado 
- [x] @catarinapm

### CODE REVIEW

File: fe/app/components/EventGrid/EventGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import RoundArrowButton from "@components/RoundArrowButton/RoundArrowButton";
+import { useCallback, useEffect, useState } from "react";
+import EventElement from "../EventElement/EventElement";
+import { EventElementProps } from "../EventElement/EventElement.models";
+import { useTranslations } from "next-intl";
+
+const EventGrid = ({ events }: { events: EventElementProps[] }) => {
```
Comment (danielaprado) [inline fe/app/components/EventGrid/EventGrid.tsx:7]: I know it has only one prop but shouldn't we still have a `models.ts` file to go along with the standard we are using on the rest of the components?

File: fe/app/components/EventGrid/EventGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  const [numOfElems, setNumOfElems] = useState(0);
+
+  const getNumberOfElements = useCallback(() => {
+    if (innerWidth > 0 && innerWidth < 640) {
+      return 4;
+    } else if (innerWidth >= 640 && innerWidth < 768) {
+      return 3;
+    } else if (innerWidth >= 768 && innerWidth < 1024) {
+      return 3;
+    } else if (innerWidth >= 1024 && innerWidth < 1280) {
```
Comment (danielaprado) [inline fe/app/components/EventGrid/EventGrid.tsx:22]: Also these two?

File: fe/app/components/EventGrid/EventGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+  const [showAll, setShowAll] = useState(false);
+  const [loadMore, setLoadMore] = useState(false);
+  const [innerWidth, setInnerWidth] = useState(window.innerWidth);
+  const [numOfElems, setNumOfElems] = useState(0);
+
+  const getNumberOfElements = useCallback(() => {
+    if (innerWidth > 0 && innerWidth < 640) {
+      return 4;
+    } else if (innerWidth >= 640 && innerWidth < 768) {
```
Comment (danielaprado) [inline fe/app/components/EventGrid/EventGrid.tsx:18]: This two can't be grouped? 🤔

File: fe/app/components/EventGrid/EventGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      return 5;
+    } else if (innerWidth >= 1280 && innerWidth < 1536) {
+      return 5;
+    } else if (innerWidth > 1536) {
+      return 9;
+    }
+    return 0;
+  }, [innerWidth]);
+
+  const init = useCallback(() => {
```
Comment (danielaprado) [inline fe/app/components/EventGrid/EventGrid.tsx:32]: Perhaps rename it to `initialState`? 😁

**Review by catarinapm — APPROVED** [review]

**Pull request merged**